tests/benchmark/utils/s2n_negotiate_client_benchmark.cc

```diff
@@ -0,0 +1,229 @@
+        print_s2n_error("Error setting host name verification function.");
+    }
+
+    unsafe_verify_data->trusted_host = host;
+
+    if (!conn_settings.insecure) {
+        const char *ca_dir = NULL;
+        if(all_suites[state.range(0)]->auth_method == S2N_AUTHENTICATION_RSA) {
+            std::string ca_file = "rsa_2048_sha384_client_cert.pem";
+            std::string pem_file_location = pem_dir + ca_file;
+            if (s2n_config_set_verification_ca_location(config, pem_file_location.c_str(), ca_dir) < 0) {
+                print_s2n_error("Error setting CA file for trust store.");
+            }
+        }
+        else {
+            std::string ca_file = "ecdsa_p256_pkcs1_cert.pem";
+            std::string pem_file_location = pem_dir + ca_file;
+            if (s2n_config_set_verification_ca_location(config, pem_file_location.c_str(), ca_dir) < 0) {
+                print_s2n_error("Error setting CA file for trust store.");
+            }
+        }
+    }
+    else {
+        GUARD_EXIT(s2n_config_disable_x509_verification(config), "Error disabling X.509 validation");
+    }
+
+    if (conn == NULL) {
+        print_s2n_error("Error getting new connection");
+        exit(1);
+    }
+
+    GUARD_EXIT(s2n_connection_set_config(conn, config), "Error setting configuration");
+
+    GUARD_EXIT(s2n_connection_set_client_auth_type(conn, S2N_CERT_AUTH_OPTIONAL),
+               "Error setting ClientAuth optional");
+
+    if (conn_settings.use_corked_io) {
+        GUARD_EXIT(s2n_connection_use_corked_io(conn), "Error setting corked io");
+    }
+
+    for (size_t i = 0; i < WARMUP_ITERS; i++) {
+        client_handshake(state, true, conn);
+    }
+    for (auto _ : state) {
+        state.PauseTiming();
+        client_handshake(state, false, conn);
+    }
+    free(unsafe_verify_data);
+    GUARD_EXIT(s2n_config_free(config), "Error freeing configuration");
+    GUARD_EXIT(s2n_connection_free(conn), "Error freeing connection");
+}
+
+int start_negotiate_benchmark_client(int argc, char** argv) {
+    int use_corked_io = 0;
+    int insecure = 1;
+    conn_settings = {0};
+    char bench_format[100] = "--benchmark_out_format=";
+    char file_prefix[100];
+    size_t WARMUP_ITERS = 1;
+    size_t ITERATIONS = 1;
+    struct addrinfo hints;
+    struct addrinfo *ai_list = nullptr;
+    struct addrinfo *ai = nullptr;
+
+    argument_parse(argc, argv, use_corked_io, insecure, bench_format, file_prefix, WARMUP_ITERS, ITERATIONS);
+
+    char bench_out[100] = "--benchmark_out=client_";
+    strcat(bench_out, file_prefix);
+    argc += 2;
+
+    std::vector<char*> argv_bench(argv, argv + argc);
+    argv_bench.push_back(bench_out);
+    argv_bench.push_back(bench_format);
+    argv_bench.push_back(nullptr);
+
+    conn_settings.use_corked_io = use_corked_io;
+    conn_settings.insecure = insecure;
+
+    s2n_init();
+
+    GUARD_EXIT(setup_socket(hints, ai_list, ai), "setup failed");
```

Comment:
These parameters can be moved, also add to the error message what part of the setup failed.